File: JUNHEE/프로그래머스_week14/정수 제곱근 판별.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    long long num = 1;
    while(num*num <= n){
        if(num*num == n) return (num+1)*(num+1);
        num++;
    }
    return -1;
}

File: DONGUK/프로그래머스_WEEK_14/정수 제곱근 판별.py
def solution(n):
    answer = 0
    
    a = str(n ** (1/2))
    
    a = a.split('.')
    b = a[1]
    
    if len(b) == 1 and b == '0':
        answer = (int(a[0]) + 1) ** 2
    else:
        answer = -1
    
    return answer